packages/components/src/unit-control/utils.ts

```diff
@@ -178,17 +178,18 @@ export function hasUnits( units: WPUnitControlUnitList ): boolean {
  * @return The extracted number and unit.
  */
 export function parseUnit(
-	initialValue: Value,
+	initialValue: Value | undefined,
 	units: WPUnitControlUnitList = ALL_CSS_UNITS
-): [ Value, string ] {
+): [ Value, string | undefined ] {
 	const value = String( initialValue ).trim();
 
 	let num: Value = parseFloat( value );
 	num = isNaN( num ) ? '' : num;
 
-	const unitMatch = value.match( /[\d.\-\+]*\s*(.*)/ )[ 1 ];
+	const unitMatch = value.match( /[\d.\-\+]*\s*(.*)/ );
 
-	let unit = unitMatch !== undefined ? unitMatch : '';
+	let unit: string | undefined =
+		unitMatch?.[ 1 ] !== undefined ? unitMatch[ 1 ] : '';
```

Comment:
The logic here needed to be updated because `value.match()` can return `null` is there is no match, so I've added in optional chaining before attempting to access the index.

Reply:
Thank for the explanation.

I just don't get why `unit` is typed as `string | undefined` — shouldn't it be guaranteed to be a `string` (given the fallback to `''`)?